hermes/probe/monitor.go

```diff
@@ -143,22 +142,22 @@ func (mp *MonitorProbe) runProbe(ctx context.Context, metricChan chan<- *cpmetri
 				mp.logger.Errorf(err.Error())
 				return
 			}
-			probeCtx, _ := context.WithDeadline(ctx, time.Now().Add(mp.opts.Timeout)) //TODO: Change to mp.Config. ...
+			probeCtx, _ := context.WithDeadline(ctx, time.Now().Add(time.Duration(mp.config.GetIntervalSec())))
```

Comment:
The time.Duration unit is nanoseconds. To fix this, multiple `mp.config.GetIntervalSec()` by `time.Second`.
I expect this bug would have been caught by unit-tests, as the operation would always time out.

Reply:
Ah thank you for pointing this out. Currently, the unit tests are quite sparse. I have a TODO to add more once the code is satisfactory.